torch/csrc/jit/codegen/cuda/scheduler/pointwise_utils.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <torch/csrc/jit/codegen/cuda/compute_at_map.h>
+#include <torch/csrc/jit/codegen/cuda/ir_all_nodes.h>
+#include <torch/csrc/jit/codegen/cuda/ir_utils.h>
+#include <torch/csrc/jit/codegen/cuda/scheduler/utils.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+namespace pointwise_utils {
+
+// DomainMap uses the ComputeAtMap to find a reference TensorView
+// that maps to all IterDomains in the fusion.
+class DomainMap {
+ public:
+  DomainMap(Fusion* fusion);
+  virtual ~DomainMap() = default;
```

Comment:
This would be a very useful cache entry and I do expect it to be useful in many coming scheduler variants.

Reply:
Also just wondering, why does this need to be virtual? Meanwhile, could you add an interface function that exposes the underlying `ca_map_` ? That'd be very helpful in many scenarios.

Reply:
The compute at map seems a bit dangerous to expose directly during scheduling as it will become out of date during scheduling. Though if we need the unscheduled ca_map_ that could make sense.

Reply:
Yes was hoping that this would cover the unscheduled ca_map usage for the heuristic cache entry.  If this domain map is also used in scheduling phase, it'd need to be guarded.

Reply:
We could just revisit when other schedulers need to use `ca_map_` as well.

Reply:
I think data caching is not available in the scheduling phase, so whenever you need it in scheduling, you rebuild it.

Reply:
Added a
```C++
  const ComputeAtMap &getComputeAtMap() const {
    return ca_map_;
  }
```

Reply:
> Also just wondering, why does this need to be virtual?

I would expect different schedulers to subclass this, but uses the same cache entry for caching. For this to work, I would need `dynamic_cast`, which requires the base class to be polymorphic.